### PR TITLE
Document Ghc 69925

### PR DESCRIPTION
**Pull request by PPKFS**
I think my wording over unboxed types may need a second pair of eyes but I think it was alright

### CODE REVIEW

File: message-index/messages/GHC-69925/index.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Illegal unboxed string literal in pattern
+summary: Illegal unboxed string literal in pattern
+severity: error
+introduced: 9.6.1
+---
+
+When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. `"foo"#` is of type `Addr#` -- that is, "An arbitrary machine address assumed to point outside the garbage-collected heap.".
```
Comment (david-christiansen) [inline message-index/messages/GHC-69925/index.md:8]:
```suggestion
When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. Unboxed string literals are essentially C strings allocated outside of the Haskell heap, and they evaluate to pointers to the resulting strings. Thus,`"foo"#` is of type `Addr#`. While `eqAddr#` can be used to compare these pointers, two identically-written unboxed string literals typically do not point the same address.".
```

Comment (david-christiansen) [reply message-index/messages/GHC-69925/index.md:8]: I think the fix in the example should be to use boxed string literals instead :-)

**Comment by PPKFS** [comment]
updated the fix to use boxed literals

File: message-index/messages/GHC-69925/example1/after/IllegalUnboxedString.hs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module IllegalUnboxedString where
+
+g :: String -> Bool
+g y = (y == "a")
```
Comment (david-christiansen) [inline message-index/messages/GHC-69925/example1/after/IllegalUnboxedString.hs:4]:
```suggestion
g y = case y of
    "a" -> True
    _ -> False
```

For consistency with the other version.

File: message-index/messages/GHC-69925/index.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+title: Illegal unboxed string literal in pattern
+summary: Illegal unboxed string literal in pattern
+severity: error
+introduced: 9.6.1
+---
+
+When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. Unboxed string literals are essentially C strings allocated outside of the Haskell heap, and they evaluate to pointers to the resulting strings. Thus,`"foo"#` is of type `Addr#`. While `eqAddr#` can be used to compare these pointers, two identically-written unboxed string literals typically do not point the same address.".
```
Comment (david-christiansen) [inline message-index/messages/GHC-69925/index.md:8]:
```suggestion
When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. Unboxed string literals are essentially C strings allocated outside of the Haskell heap, and they evaluate to pointers to the resulting strings. Thus,`"foo"#` is of type `Addr#`. While `eqAddr#` can be used to compare these pointers, two identically-written unboxed string literals typically do not point the same address.
```

File: message-index/messages/GHC-69925/index.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+Instead, consider using the unboxed function that implements equality (`eqAddr#`) which can then be used in combination with `if .. then .. else` via `isTrue#`.
```
Comment (david-christiansen) [inline message-index/messages/GHC-69925/index.md:10]:
```suggestion
```

File: message-index/messages/GHC-69925/index.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+---
+title: Illegal unboxed string literal in pattern
+summary: Illegal unboxed string literal in pattern
+severity: error
+introduced: 9.6.1
+---
+
+When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. Unboxed string literals are essentially C strings allocated outside of the Haskell heap, and they evaluate to pointers to the resulting strings. Thus,`"foo"#` is of type `Addr#`. While `eqAddr#` can be used to compare these pointers, two identically-written unboxed string literals typically do not point the same address.
```
Comment (elland) [inline message-index/messages/GHC-69925/index.md:8]: How about cross-referencing docs for magic hash?

```suggestion
When pattern matching, unboxed string literals (with the [MagicHash](https://downloads.haskell.org/ghc/latest/docs/html/users_guide/exts/magic_hash.html?highlight=magichash
) postfix) are not allowed. Unboxed string literals are essentially C strings allocated outside of the Haskell heap, and they evaluate to pointers to the resulting strings. Thus,`"foo"#` is of type `Addr#`. While `eqAddr#` can be used to compare these pointers, two identically-written unboxed string literals typically do not point the same address.
```

**Review by elland — APPROVED** [review]

**Pull request merged**